app/main.py

```diff
@@ -2,0 +1,30 @@
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        self.__class__.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, animal: Animal) -> None:
+        if not isinstance(animal, Carnivore) and not animal.hidden:
+            animal.health -= 50
+            if animal.health <= 0:
+                self.__class__.alive.remove(animal)
```

Comment:
Removing an animal from `self.__class__.alive` will affect all instances of `Animal` and its subclasses. Ensure this is the desired behavior, as it might lead to unexpected results if different subclasses are meant to have separate `alive` lists.